feat: 근로계약서 OCR 검증 구현

djlim00 opened the pull request:
## 📚 개요

- 파이썬 서버에 근로계약서 파일과 user.name, companyName을 통해서 검증을 하는 기능입니다.

## 🕐 리뷰 예상 시간

> 30m

## ❗❗ 중요한 변경점(Option)

- Arabyte-OCR도 참고해주시면 감사하겠습니다.
- 안드에서 user.name과 companyName, 근로계약서 파일을 받아서 OCR검증 서버에 요청을 보냅니다.
- 검증 결과를 반환합니다.

djlim00 committed:
feat: 근로계약서 OCR 검증 구현

- 파이썬 서버에 근로계약서 파일과 user.name, companyName을 통해서 검증을 하는 기능입니다.

djlim00 committed:
feat: 근로계약서 OCR 검증 구현

- 파이썬 서버에 근로계약서 파일과 user.name, companyName을 통해서 검증을 하는 기능입니다.

djlim00 committed:
update: baseUrl을 application.yml에서 관리하도록 수정

djlim00 committed:
update: baseUrl을 application.yml에서 관리하도록 수정

Ji-InPark committed:
Update GetReviewsResponse.kt

Ji-InPark committed:
Update Location.kt

Ji-InPark commented on src/main/resources/application.yml:
```diff
@@ -28,4 +28,8 @@ spring:
\ No newline at end of file
+    operations-sorter: method
+
+ocr:
+  server:
+    base-url: http://localhost:8000
```
이 부분은 application-local에 적어주셔야 해요

djlim00 replied:
확인했습니다.

Ji-InPark commented on src/main/kotlin/com/arabyte/arabyteapi/domain/mypage/service/ContractService.kt:
```diff
@@ -0,0 +1,54 @@
+        val userName = user.username
+        
+        val originalFilename = file.originalFilename ?: "contract.png"
+        val extension = originalFilename.substringAfterLast('.', "png")
```
```suggestion
        val extension = StringUtils.getFilenameExtension(originalFilename)
```

Ji-InPark commented on src/main/kotlin/com/arabyte/arabyteapi/global/configuration/RetrofitConfiguration.kt:
```diff
@@ -11,7 +13,8 @@ import java.util.concurrent.TimeUnit
 class RetrofitConfiguration(
-    private val objectMapper: ObjectMapper
+    private val objectMapper: ObjectMapper,
+    @Value("\${ocr.server.base-url}") private val baseUrl: String
```
```suggestion
    @Value("\${ocr.server.base-url}") 
    private val baseUrl: String
```

djlim00 replied:
넵 확인했습니다~ 반영할게요

Ji-InPark committed:
Add user id to review response

Ji-InPark committed:
Add feild and class about review helpful

Ji-InPark committed:
Add implement review helpful logics

Ji-InPark committed:
Edit apply review

Ji-InPark committed:
Merge pull request #29 from TEAM-Arabyte/feature/add-review-helpful-feature

ADD - 리뷰 도움돼요 기능 추가

Ji-InPark committed:
Update 03-01-changelog.xml

djlim00 committed:
Update src/main/kotlin/com/arabyte/arabyteapi/domain/mypage/service/ContractService.kt

Co-authored-by: Ji-In Park <81283634+Ji-InPark@users.noreply.github.com>

djlim00 committed:
Update src/main/kotlin/com/arabyte/arabyteapi/domain/mypage/service/ContractService.kt

Co-authored-by: Ji-In Park <81283634+Ji-InPark@users.noreply.github.com>

djlim00 committed:
update: 리뷰 내역 반영

- 줄 바꿈, 사용함수와 같은 부분들 리뷰 내용 반영했습니다.
- local ocr서버 설정을 application-local.yml로 이동했습니다.

djlim00 committed:
update: 리뷰 내역 반영

- 줄 바꿈, 사용함수와 같은 부분들 리뷰 내용 반영했습니다.
- local ocr서버 설정을 application-local.yml로 이동했습니다.

Ji-InPark reviewed (commented):
일단 ocr 배포하고 baseUrl도 추가하고 머지하는걸로 하죠
먼저 rebase로 컨플릭해결해주세요~

Ji-InPark committed:
Add implement company

Ji-InPark committed:
Add querydsl dependency

Ji-InPark committed:
Create CustomReviewRepository.kt

Ji-InPark committed:
Add filter review logic

Ji-InPark committed:
Add review helpful response

Ji-InPark committed:
Merge pull request #30 from TEAM-Arabyte/feature/implement-company

ADD - company 구현 및 적용

Ji-InPark committed:
Merge pull request #31 from TEAM-Arabyte/feature/add-review-filetering

ADD - 리뷰 필터링 구현 및 도움돼요 응답 추가

Ji-InPark committed:
Create 05-02-changelog.xml

Ji-InPark committed:
Add companyName to get review response

Ji-InPark committed:
Add discord api and service

Ji-InPark committed:
Add report feature

Ji-InPark committed:
Edit report layout

Ji-InPark committed:
Merge pull request #33 from TEAM-Arabyte/feature/add-report

ADD - 신고하기 기능 구현

djlim00 committed:
Merge branch 'feature/mypage-ocr' of https://github.com/TEAM-Arabyte/Arabyte-API into feature/mypage-ocr

djlim00 committed:
udpate: 설정 파일 수정
